File: frontend/src/admin/js/deleteAllCards.js
import axios from 'axios';
import Swal from 'sweetalert2';
import { domain } from '../../utils';

const handleClearButtonClick = async (setCards) => {
    try {
        const { value: password } = await Swal.fire({
            title: 'Enter Admin Password',
            input: 'password',
            inputPlaceholder: 'Admin Password',
            confirmButtonText: 'Login',
            showCancelButton: true,
            inputValidator: (value) => {
                if (!value) {
                    return 'Password is required!';
                }
            },
        });

        if (!password) {
            return;
        }
        const response = await axios.delete(`${domain}/api/delete-all-card/`, {
            data: { password }
        });

        if (response.status === 204) {
            setCards([]);
            Swal.fire('Success!', 'All cards have been deleted.', 'success');
        } else {
            Swal.fire('Error', 'An unexpected error occurred.', 'error');
        }
    } catch (error) {
        if (error.response && error.response.status === 403) {
            Swal.fire({
                icon: 'error',
                title: 'Access Denied',
                text: error.response.data?.message || 'Invalid password.',
            });
        } else if (error.response) {
            Swal.fire({
                icon: 'error',
                title: 'Failed!',
                text: error.response.data?.message || 'Something went wrong on the server.',
            });
        } else if (error.request) {
            Swal.fire({
                icon: 'error',
                title: 'Failed!',
                text: 'Network error. Please check your connection.',
            });
        } else {
            Swal.fire({
                icon: 'error',
                title: 'Failed!',
                text: `Unexpected error: ${error.message}`,
            });
        }
    }
};

export default handleClearButtonClick;

File: backend/api/views/cards/delete_all_cards.py
from rest_framework import status
from rest_framework.decorators import api_view
from rest_framework.response import Response
from ...models import Card
from django.contrib.auth import get_user_model


@api_view(['DELETE'])
def card_delete_all(request):
    if request.method == 'DELETE':
        try:
            password = request.data.get('password')
            
            if not password:
                return Response({'error': 'Password is required'}, status=status.HTTP_400_BAD_REQUEST)
            user = get_user_model().objects.filter(is_superuser=True).first()

            if user:
                if user.check_password(password):
                    if not Card.objects.exists():
                        return Response({"message": "No Cards to Delete"}, status=status.HTTP_404_NOT_FOUND)
                    Card.objects.all().delete()
                    return Response({"message": "All cards have been deleted"}, status=status.HTTP_204_NO_CONTENT)
                else:
                    return Response({"error": "Invalid password"}, status=status.HTTP_403_FORBIDDEN)
            else:
                return Response({"error": "No admin user found"}, status=status.HTTP_404_NOT_FOUND)
        
        except Exception as e:
            return Response({"error": str(e)}, status=status.HTTP_400_BAD_REQUEST)
